factcast-snapshotcache-jdbc/src/main/java/org/factcast/core/snap/jdbc/JdbcSnapshotCache.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright © 2017-2020 factcast.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.factcast.core.snap.jdbc;
+
+import com.google.common.collect.Sets;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.time.LocalDate;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import javax.sql.DataSource;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.factcast.core.snap.Snapshot;
+import org.factcast.core.snap.SnapshotId;
+import org.factcast.factus.snapshot.SnapshotCache;
+
+@Slf4j
+public class JdbcSnapshotCache implements SnapshotCache {
+  public final String queryStatement;
+  public final String mergeStatement;
+  public final String deleteStatement;
+  public final String cleanupStatement;
+  private final DataSource dataSource;
+
+  public JdbcSnapshotCache(JdbcSnapshotProperties properties, DataSource dataSource) {
+    this.dataSource = dataSource;
+
+    queryStatement =
+        "SELECT * FROM " + properties.getSnapshotsTableName() + " WHERE key = ? AND uuid = ?";
```

Comment:
minor, maybe we can have a local variable instead of calling the same getter 4 times in a row